cli/flox-watchdog/src/logger.rs

```diff
@@ -40,3 +48,245 @@ pub(crate) fn init_logger(file_path: &Option<PathBuf>) -> Result<(), anyhow::Err
         .init();
     Ok(())
 }
+
+/// Starts a background thread which emits a log entry at an interval. This is
+/// used as an indication of whether a watchdog's log file can be garbage
+/// collected. The thread will run until the watchdog exits.
+pub(crate) fn spawn_heartbeat_log() {
+    spawn(move || loop {
+        debug!("still watching, woof woof");
+        sleep(HEARTBEAT_INTERVAL);
+    });
+}
```

Comment:
suggestion: add compile time check that HEARTBEAT_INTERVAL falls into KEEP_WATCHDOG_DAYS days.
The curent `#[test]` method can be dropped in favor


```suggestion
    /// Assert that HEARTBEAT_INTERVAL falls in the range of KEEP_WATCHDOG_DAYS at compile time.
    const _: () = assert!(
        HEARTBEAT_INTERVAL.as_secs() < duration_from_days(KEEP_WATCHDOG_DAYS).as_secs(),
        "HEARTBEAT_INTERVAL must be less than `KEEP_WATCHDOG_DAYS` days"
    );
```

Reply:
Done 👍🏻